assets/js/base/context/cart-checkout/checkout-state/types.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Internal dependencies
+ */
+import type { emitterCallback } from '../../shared/event-emit';
+import { STATUS } from './constants';
+import type { ActionType } from './actions';
+
+interface ErrorResponse {
+	code: string;
+	message: string;
+	data: {
+		status: number;
+	};
+}
+
+interface SuccessResponse {
+	// eslint-disable-next-line camelcase
+	payment_result: {
+		// eslint-disable-next-line camelcase
+		payment_status: 'success' | 'failure' | 'pending' | 'error';
+		// eslint-disable-next-line camelcase
+		payment_details: Record< string, string > | Record< string, never >;
+		// eslint-disable-next-line camelcase
+		redirect_url: string;
+	};
+}
+
+export type CheckoutResponse = SuccessResponse | ErrorResponse;
+
+export interface PaymentResultDataType {
+	message: string;
+	paymentStatus: string;
+	paymentDetails: Record< string, string > | Record< string, never >;
+	redirectUrl: string;
+}
+
+export type CheckoutStateContextState = {
+	redirectUrl: string;
+	status: STATUS;
+	hasError: boolean;
+	calculatingCount: number;
+	orderId: number;
+	orderNotes: string;
+	customerId: number;
+	shouldCreateAccount: boolean;
+	processingResponse: PaymentResultDataType | null;
+};
+
+export type CheckoutStateContextType = {
+	// Checkout state provider dispatch function.
+	dispatch: React.Dispatch< ActionType >;
+	// Submits the checkout and begins processing.
+	onSubmit: () => void;
```

Comment:
I've left this in place because it's exposed to Payment Methods.

Reply:
Potentially, the `usePaymentMethodIntegrations` hook _could_ just pull in the dispatch functions and expose them though right?